Modules/System/Cryptography Management/src/DESCryptoServiceProvider.al

```diff
@@ -0,0 +1,68 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+/// <summary>
+/// Provides helper functions for the Advanced Encryption Standard.
+/// </summary>
+
+codeunit 50102 "DESCryptoServiceProvider"
+{
+    Access = Public;
+
+    /// <summary>
+    /// Encrypts text with DESCryptoServiceProvider
+    /// </summary>
+    /// <param name="Password">Represents the password to be used to initialize a new instance of Rfc2898DeriveBytes</param>
+    /// <param name="Salt">Represents the salt to be used to initialize a new instance of Rfc2898DeriveBytes</param>
+    /// <param name="VarInput">Represents the input text to be encrypted</param>
+    procedure EncryptTextWithDESCryptoServiceProvider(VarInput: Text; Password: Text; Salt: Text) VarOutput: Text
+    var
+        CryptographyManagementImpl: Codeunit "Cryptography Management Impl.";
+    begin
+        VarOutput := CryptographyManagementImpl.EncryptTextWithDESCryptoServiceProvider(VarInput, Password, Salt);
+    end;
+
+    /// <summary>
+    /// Decrypts text with DESCryptoServiceProvider
+    /// </summary>
+    /// <param name="Password">Represents the password to be used to initialize a new instance of Rfc2898DeriveBytes</param>
+    /// <param name="Salt">Represents the salt to be used to initialize a new instance of Rfc2898DeriveBytes</param>
+    /// <param name="VarInput">Represents the input text to be encrypted</param>
+    procedure DecryptTextWithDESCryptoServiceProvider(VarInput: Text; Password: Text; Salt: Text) VarOutput: Text
+    var
+        CryptographyManagementImpl: Codeunit "Cryptography Management Impl.";
+    begin
+        VarOutput := CryptographyManagementImpl.DecryptTextWithDESCryptoServiceProvider(VarInput, Password, Salt);
+    end;
+
+
+    /// <summary>
+    /// Encrypts data in stream with DESCryptoServiceProvider
+    /// </summary>
+    /// <param name="Password">Represents the password to be used to initialize a new instance of Rfc2898DeriveBytes</param>
+    /// <param name="Salt">Represents the salt to be used to initialize a new instance of Rfc2898DeriveBytes</param>
+    /// <param name="InputInstream">Represents the input instream data to be encrypted</param>
+    /// <param name="OutputOutstream">Represents the output instream encrypted data </param>
```

Comment:
The output instream encrypted data.

Reply:
Hi Brent, that's exactly the same as is already standing there? Or should I remove the "Represents"?